src/main/java/org/opensearch/security/securityconf/impl/v7/ConfigV7.java

```diff
@@ -248,6 +249,24 @@ public AuthFailureListener(ConfigV6.AuthFailureListener v6) {
             this.max_tracked_clients = v6.max_tracked_clients;
         }
 
+        public AuthFailureListener(
```

Comment:
I wonder if it makes sense to separate out IPAuthFailureListener and UsernameAuthFailureListener. They both could inherit from an AbstractAuthFailureListener and re-use the parent's constructor for common attributes.